### PR TITLE
don't print <exception entry> more thn once per frame

### LINKED ISSUE #59: `<exception entry>` printed twice on UDF
STR
``` rust
#[cortex_m_rt::entry]
fn main() -> ! {
    cortex_m::asm::udf();
}
```

``` text
$ cargo rb abort
stack backtrace:
   0: 0x000003e0 - HardFaultTrampoline
      <exception entry>
      <exception entry>
   1: 0x00000140 - __udf
   2: 0x00000118 - cortex_m::asm::udf
   3: 0x0000012c - abort::__cortex_m_rt_main
   4: 0x00000122 - main
   5: 0x000000fa - Reset
```

### PR DESCRIPTION
**Pull request by japaric**
as far as I could test `HardFaultTrampoline` always has a special LR value so we always enter the block at line 797 and print `<exception entry>`

fixes #59

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -747,7 +747,6 @@ fn backtrace(
         if top_exception.is_none() {
```
Comment (japaric) [inline src/main.rs:747]: I'm not sure the code above this line applies anymore. the code inside the if checks for overflow but doesn't do special logic around unwinding the frame. The frame is unwound normally using the logic in line 797.

probably the code around this can be simplified now that we put a breakpoint on HardFaultTrampoline

Comment (jonas-schievink) [reply src/main.rs:747]: Yeah, seems like a good idea. I do find the code in probe-run to be a bit hard to follow sometimes.

**Review by jonas-schievink — APPROVED** [review]
